src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -116,21 +72,31 @@ private <T> List<T> runSnapshotBuilderQuery(
 
   public SnapshotBuilderGetConceptsResponse getConceptChildren(
       UUID datasetId, Integer conceptId, AuthenticatedUserRequest userRequest) {
-    // TODO: Build real query - this should get the name and ID from the concept table, the count
-    // from the occurrence table, and the existence of children from the concept_ancestor table.
     Dataset dataset = datasetService.retrieve(datasetId);
-
-    String cloudSpecificSQL = buildConceptChildrenQuery(dataset, conceptId, userRequest);
-
+    TableNameGenerator tableNameGenerator = getTableNameGenerator(dataset, userRequest);
+    String cloudSpecificSql = buildConceptChildrenQuery(conceptId, tableNameGenerator);
     List<SnapshotBuilderConcept> concepts =
         runSnapshotBuilderQuery(
-            cloudSpecificSQL,
+            cloudSpecificSql,
             dataset,
             AggregateBQQueryResultsUtils::aggregateConceptResults,
             AggregateSynapseQueryResultsUtils::aggregateConceptResult);
     return new SnapshotBuilderGetConceptsResponse().result(concepts);
   }
 
+  private TableNameGenerator getTableNameGenerator(
+      Dataset dataset, AuthenticatedUserRequest userRequest) {
+    CloudPlatformWrapper platform = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    if (platform.isGcp()) {
+      return BigQueryVisitor.bqTableName(datasetService.retrieveModel(dataset, userRequest));
+    } else if (platform.isAzure()) {
```

Comment:
This is really easy to follow! I think the `else if` can be replaced with `if` though?

Reply:
I don't know if this adds extra lines of code but thoughts on a switch statement?

Reply:
`else if` has the benefit of short-circuiting the execution of the clause (if the first condition is true, we don't want to test the second condition), so I would lean that way.